components/style/properties/longhand/box.mako.rs

```diff
@@ -1388,34 +1410,66 @@ ${helpers.predefined_type("scroll-snap-coordinate",
                 &name,
                 "matrix" => {
                     try!(input.parse_nested_block(|input| {
-                        let values = try!(input.parse_comma_separated(|input| {
-                            specified::parse_number(input)
-                        }));
-                        if values.len() != 6 {
-                            return Err(())
+                        let mut values = vec![];
+                        let mut lengths = vec![];
+                        // Consume first number
+                        values.push(specified::parse_number(input)?);
+                        // Parse other 5 number/LengthOrPercentageOrNumber
+                        for i in 0..5 {
+                            input.expect_comma()?;
+                            if i < 3 {
+                                values.push(specified::parse_number(input)?);
+                            } else {
+                                // -moz-transform accepts LengthOrPercentageOrNumber in the nondiagonal
+                                // homogeneous components. transform accepts only number.
+                                if prefixed {
+                                    lengths.push(LoPoNumber::parse(context, input)?)
+                                } else {
+                                    lengths.push(Either::Second(specified::parse_number(input)?));
+                                }
+                            }
                         }
+
                         let matrix = SpecifiedMatrix {
                             m11: values[0],        m12: values[1],        m13: Number::new(0.0), m14: Number::new(0.0),
                             m21: values[2],        m22: values[3],        m23: Number::new(0.0), m24: Number::new(0.0),
                             m31: Number::new(0.0), m32: Number::new(0.0), m33: Number::new(1.0), m34: Number::new(0.0),
-                            m41: values[4],        m42: values[5],        m43: Number::new(0.0), m44: Number::new(1.0),
+                            m41: lengths[0].clone(), m42: lengths[1].clone(),
```

Comment:
I'm not happy with these indentations but I couldn't find a better way to indent this :(